src/sentry/search/moresnuba/backend.py

```diff
@@ -0,0 +1,149 @@
+    )
+    sort_strategies = {
+        # TODO: If not using environment filters, could these sort methods use last_seen and first_seen from groups instead? so only add prefix conditionally?
+        "date": "events.last_seen",
+        "freq": "times_seen",
+        "new": "events.first_seen",
+        "priority": "priority",
+    }
+
+    aggregation_defs = {
+        "times_seen": ["count()", ""],
+        "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+        "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+        # https://github.com/getsentry/sentry/blob/804c85100d0003cfdda91701911f21ed5f66f67c/src/sentry/event_manager.py#L241-L271
+        "priority": ["toUInt64(plus(multiply(log(times_seen), 600), `events.last_seen`))", ""],
+        # Only makes sense with WITH TOTALS, returns 1 for an individual group.
+        "total": ["uniq", "events.issue"],
+    }
+
+    def __init__(self):
+        self.issue_only_fields.discard("active_at")
+        self.issue_only_fields.discard("first_seen")
+        self.issue_only_fields.discard("first_release")
+        self.issue_only_fields.discard("status")
+
+    # def query(
+    #     self,
+    #     projects,
+    #     environments=None,
+    #     sort_by="date",
+    #     limit=100,
+    #     cursor=None,
+    #     count_hits=False,
+    #     paginator_options=None,
+    #     search_filters=None,
+    #     date_from=None,
+    #     date_to=None,
+    # ):
+    #     from sentry.models import Group, GroupStatus, GroupSubscription
+    #     alias_variable_definitions();
+    #     return super(MoreSnubaSearchBackend,self).query()
+
+    # def alias_variable_definitions():
+    #    pass
+
+    def get_queryset_builder_conditions(
+        self,
+        projects,
+        environments=None,
+        sort_by="date",
+        limit=100,
+        cursor=None,
+        count_hits=False,
+        paginator_options=None,
+        search_filters=None,
+        date_from=None,
+        date_to=None,
+    ):
+        # We override this function to remove status and active_at
+        qs_builder_conditions = super(MoreSnubaSearchBackend, self).get_queryset_builder_conditions(
+            projects,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            date_from,
+            date_to,
+        )
+
+        del qs_builder_conditions["status"]
+        del qs_builder_conditions["active_at"]
+
+        return qs_builder_conditions
+
+    def build_environment_and_release_queryset(
+        self, projects, group_queryset, environments, search_filters
+    ):
+        # Override the base function, which filters the group_queryset,
+        # and do no postgres filter building for first_release and first_seen
+        # Just return it unmodified. We do this filter in snuba now.
+        return group_queryset
```

Comment:
How do you plan to run a different logic between queries that will need postgres and queries which will not?
Is it supposed to happen here by adding different conditions to the group_queryset ?

Reply:
Yes, as we remove conditions we have inthe group_queryset, we can reduce postgres query load. In moresnuba::get_queryset_builder_conditions, we call the original snuba function and then delete a couple of its keys (the ones we can query in snuba: status and active_at. We can also search on groups.last_seen and groups.first_seen instead of using the events aggregate if there is no environment filter). Those don't get built into the queryset now.

We can continue to improve moresnuba over time. The original point, as I understand it, was not to use it for querying on the fields we now have in snuba. I'm happy to keep optimizing this :)

This PR is to get the refactored snuba backed into master before merge conflicts are a headache.